modules/tensor_mechanics/src/materials/ScalarMaterialDamage.C

```diff
@@ -0,0 +1,61 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ScalarMaterialDamage.h"
+#include "Conversion.h"
+
+registerMooseObject("TensorMechanicsApp", ScalarMaterialDamage);
+
+template <>
+InputParameters
+validParams<ScalarMaterialDamage>()
+{
+  InputParameters params = validParams<DamageBase>();
+  params.addClassDescription(
+      "Scalar damage model for which the damage is prescribed by another material");
+  params.addParam<bool>(
+      "use_old_damage",
+      false,
+      "Whether to use the damage index from the previous step in the stress computation");
+  params.addRequiredParam<MaterialPropertyName>("damage_index",
+                                                "Name of the material property containing the "
+                                                "damage index, which goes from 0 (undamaged) to 1 "
+                                                "(fully damaged)");
+  return params;
+}
+
+ScalarMaterialDamage::ScalarMaterialDamage(const InputParameters & parameters)
+  : DamageBase(parameters),
+    _use_old_damage(getParam<bool>("use_old_damage")),
+    _damage_index(
+        _use_old_damage
+            ? getMaterialPropertyOldByName<Real>(getParam<MaterialPropertyName>("damage_index"))
+            : getMaterialPropertyByName<Real>(getParam<MaterialPropertyName>("damage_index")))
+{
+}
+
+void
+ScalarMaterialDamage::updateDamage()
+{
+  if (_damage_index[_qp] < 0.0 || _damage_index[_qp] > 1.0)
+    mooseError("damage_index must be between 0 and 1. Current value is: " +
+               Moose::stringify(_damage_index[_qp]));
+}
+
+void
+ScalarMaterialDamage::updateStressForDamage(RankTwoTensor & stress_new)
+{
+  stress_new *= (1.0 - _damage_index[_qp]);
+}
+
+void
+ScalarMaterialDamage::updateJacobianMultForDamage(RankFourTensor & jacobian_mult)
+{
+  jacobian_mult *= (1.0 - _damage_index[_qp]);
```

Comment:
What happens if ``_damage_index[_qp]`` is exactly 1? Can MOOSE find a solution? I'm asking because I noticed that the tests you have set up don't go up to 1.

If MOOSE can't find a solution when ``_damage_index[_qp]`` is exactly 1, then maybe there is a need for an additional parameter that would prevent the jacobian to go to 0 (for example a ``_residual_stiffness_fraction`` (strictly positive) so that we have ``jacobian_mult *= std::max((1.0 - _damage_index[_qp]), _residual_stiffness_fraction);``